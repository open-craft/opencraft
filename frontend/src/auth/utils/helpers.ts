import { V2Api } from 'global/api';

export const checkAuthAndRefreshToken = async () => {
  const refreshToken = localStorage.getItem('token_refresh');
  let response = null;

  if (refreshToken) {
    try {
      response = await V2Api.authRefreshCreate({
        data: { refresh: refreshToken }
      });
    } catch (e) {
      return false;
    }
  }

  if (response) {
    localStorage.setItem('token_access', response.access);
    return true;
  }

  return false;
};
