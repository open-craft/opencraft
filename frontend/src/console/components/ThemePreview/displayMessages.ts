const messages = {
  previewNotice: {
    defaultMessage:
      'This is only a preview. Publish changes to update the live site',
    description: 'Notifies the user regarding the preview functionality'
  }
};

export default messages;
