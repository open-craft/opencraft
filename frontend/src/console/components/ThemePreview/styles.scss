@import '~styles/theme';

.preview-header {
  position:relative;
  height: 60px;
  z-index: 2;
}

.notice-container {
  height: 100%;

  color: $primary-1;
  margin-top: 20px;
  margin-bottom: auto;
  font-size: 14px;
  font-weight: normal;
  font-stretch: normal;
  font-style: $chivo-font;
  line-height: normal;
  letter-spacing: normal;
  text-align: right;
}

.preview-container {
  z-index: 1;
  position: relative;
  padding: 0;

  .row {
    padding: 0;

    .side-buttons {
      padding: 0 15px 0 0;

    }
  }
}
