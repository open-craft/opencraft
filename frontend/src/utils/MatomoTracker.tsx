import React, { ReactNode } from 'react';
import {
  createInstance,
  MatomoProvider,
  useMatomo
} from '@datapunt/matomo-tracker-react';
import {
  MATOMO_ALIAS_DOMAIN,
  MATOMO_BASE_URL,
  MATOMO_MY_DOMAIN,
  MATOMO_SITE_ID
} from '../global/constants';

interface MatomoTrackerProps {
  children: ReactNode;
}

interface MatomoAdditionalConfiguration {
  [propName: string]: any;
}

interface MatomoConfiguration {
  urlBase: string;
  siteId: number;
  configurations?: MatomoAdditionalConfiguration;
}

interface MatomoUserIdTrackerProps {
  userId: string;
}

/*
 * This wrapper component initializes the Matomo tracking by wrapping the children
 * elements, if the tracking is enabled by providing a valid value for the 'MATOMO_BASE_URL'
 * constant. It allows tracking page views, events and the user id, when available.
 *
 * The 'MATOMO_ALIAS_DOMAIN' constant allows the cross-domain linking of another site with
 * this site so that they are treated as the same site in Matomo.
 */
export const MatomoTracker: React.FC<MatomoTrackerProps> = (
  props: MatomoTrackerProps
) => {
  let element = props.children;
  if (MATOMO_BASE_URL) {
    const matomoConfiguration: MatomoConfiguration = {
      urlBase: MATOMO_BASE_URL,
      siteId: parseInt(MATOMO_SITE_ID, 10)
    };
    if (MATOMO_ALIAS_DOMAIN) {
      const myDomain = `*.${MATOMO_MY_DOMAIN}`;
      const aliasDomain = `*.${MATOMO_ALIAS_DOMAIN}`;
      matomoConfiguration.configurations = {
        enableCrossDomainLinking: true,
        setDomains: [myDomain, aliasDomain]
      };
    }
    element = (
      <MatomoProvider value={createInstance(matomoConfiguration)}>
        {props.children}
      </MatomoProvider>
    );
  }
  return <>{element}</>;
};

/**
 * This is a component passes the provided user id to Matomo if Matomo tracking is enabled.
 * The 'pushInstruction' function exposed by the 'useMatomo()' hook is used to do this.
 *
 * This component is needed as all the components in the registration process are classes and hence
 * the 'yseMatomo' hook cannot be directly used in those components.
 */
export const MatomoUserIdTracker: React.FC<MatomoUserIdTrackerProps> = (
  props: MatomoUserIdTrackerProps
) => {
  const { pushInstruction } = useMatomo();
  if (MATOMO_BASE_URL && props.userId) {
    pushInstruction('setUserId', props.userId);
  }
  return null;
};
