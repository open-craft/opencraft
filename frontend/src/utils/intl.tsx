import React from "react";
import { FormattedMessage, MessageDescriptor } from "react-intl";

type MessageMap = { [k: string]: Omit<MessageDescriptor, 'id'> };

interface Props<T, K extends keyof T> {
    id: K;
    messages: T;
    values?: any;
}


export const WrappedMessage = (props: Props<MessageMap, string>) => {
    const {messages, id, ...values} = props;
    return (
        <FormattedMessage  {...messages[id]} id={id} {...values}/>
    );
};
