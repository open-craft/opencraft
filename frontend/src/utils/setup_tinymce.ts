import 'tinymce/tinymce';
import 'tinymce/themes/silver';
import 'tinymce/skins/ui/oxide/skin.min.css';
import 'tinymce/skins/ui/oxide/content.min.css';
import 'tinymce/plugins/advlist';
import 'tinymce/plugins/autolink';
import 'tinymce/plugins/lists';
import 'tinymce/plugins/link';
import 'tinymce/plugins/image';
import 'tinymce/plugins/charmap';
import 'tinymce/plugins/print';
import 'tinymce/plugins/preview';
import 'tinymce/plugins/anchor';
import 'tinymce/plugins/searchreplace';
import 'tinymce/plugins/visualblocks';
import 'tinymce/plugins/code';
import 'tinymce/plugins/fullscreen';
import 'tinymce/plugins/insertdatetime';
import 'tinymce/plugins/media';
import 'tinymce/plugins/table';
import 'tinymce/plugins/paste';
import 'tinymce/plugins/help';
import 'tinymce/plugins/wordcount';
