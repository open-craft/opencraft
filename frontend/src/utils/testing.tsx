import { createMemoryHistory } from 'history';
import * as React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import * as renderer from 'react-test-renderer';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { createRootReducer } from '../global/reducers';

export const setupComponentForTesting = (
  reactContent: JSX.Element,
  storeContents = {}
) => {
  const middleware = applyMiddleware(thunk);

  const store = createStore(
    createRootReducer(createMemoryHistory()),
    storeContents,
    middleware
  );

  return renderer.create(
    <IntlProvider textComponent={React.Fragment} locale="en">
      <Provider store={store}>
        <MemoryRouter>{reactContent}</MemoryRouter>
      </Provider>
    </IntlProvider>
  );
};
