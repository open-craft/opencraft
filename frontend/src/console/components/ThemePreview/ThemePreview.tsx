import * as React from 'react';
import './styles.scss';
import { PreviewComponent } from 'console/components';
import { PreviewBox } from 'newConsole/components';
import { PreviewDropdown } from 'ui/components';
import { InstancesModel } from 'console/models';
import { Col, Row } from 'react-bootstrap';
import { connect } from 'react-redux';
import { RootState } from 'global/state';
import { WrappedMessage } from 'utils/intl';
import { updateThemeFieldValue } from 'console/actions';
import messages from './displayMessages';

interface State {
  currentPreview: string;
}
interface ActionProps {
  updateThemeFieldValue: Function;
}
interface StateProps extends InstancesModel {}
interface Props extends StateProps, ActionProps {}

export class ThemePreviewComponent extends React.PureComponent<Props, State> {
  constructor(props: Props, state: State) {
    super(props);
    this.state = {
      currentPreview: 'dashboard'
    };
  }

  private changePreview = (page: string) => {
    this.setState({
      currentPreview: page
    });
  };

  public render() {
    const instance = this.props.activeInstance;
    let themeData;

    if (instance.data && instance.data.draftThemeConfig) {
      themeData = instance.data.draftThemeConfig;
    }

    return (
      <Row className="m-0">
        <Col md="4" className="p-0">
          <div className="preview-header d-flex justify-content-start">
            <PreviewDropdown handleChange={this.changePreview} />
          </div>
        </Col>
        <Col md="8" className="p-0 align-items-start">
          <div className="notice-container d-flex justify-content-end">
            <WrappedMessage messages={messages} id="previewNotice" />
          </div>
        </Col>
        <PreviewBox>
          {themeData && themeData.version === 1 && (
            <PreviewComponent
              instanceData={instance.data!}
              currentPreview={this.state.currentPreview}
            />
          )}
        </PreviewBox>
      </Row>
    );
  }
}

export const ThemePreview = connect<
  StateProps,
  ActionProps,
  {},
  Props,
  RootState
>((state: RootState) => state.console, {
  updateThemeFieldValue
})(ThemePreviewComponent);
