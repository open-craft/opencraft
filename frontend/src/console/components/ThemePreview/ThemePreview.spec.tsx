import React from 'react';
import { setupComponentForTesting } from "utils/testing";
import { ThemePreview } from './ThemePreview';


describe("Theme preview page", function() {
  it('renders without crashing', () => {
      const tree = setupComponentForTesting(<ThemePreview />).toJSON();
      expect(tree).toMatchSnapshot();
  });

  it('Render theme settings page when instance has no theme set up.', () => {
      const tree = setupComponentForTesting(
        <ThemePreview />,
        {
          console: {
            loading: false,
            activeInstance: {
              data: {
                id: 1,
                instanceName: "test",
                subdomain: "test",
                draftThemeConfig: {},
                draftStaticContentOverrides: {
                  homepageOverlayHtml: "",
                }
              },
              deployment: {
                status: "NO_STATUS",
                undeployedChanges: [],
                deployedChanges: null,
                deploymentType: 'admin',
              }
            },
            instances: [{
              id: 1,
              instanceName: "test",
              subdomain: "test",
            }]
          }
        }
      ).toJSON();
      expect(tree).toMatchSnapshot();
  });
});
