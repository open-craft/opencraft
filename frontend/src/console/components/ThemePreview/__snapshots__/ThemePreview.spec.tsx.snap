// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme preview page Render theme settings page when instance has no theme set up. 1`] = `
<div
  className="console-page"
>
  <div
    className="d-flex justify-content-center align-middle email-activation-alert"
  >
    <i
      className="alert-icon fas fa-exclamation-triangle fa-xs"
    />
    One of your email addresses is pending confirmation. Please confirm it to begin managing your Open edX instance.
  </div>
  <div
    className="new-console-page-container"
  >
    <div
      className="new-console-page-content row"
    >
      <div
        className="container-fluid"
      >
        <div
          className="justify-content-center row"
        >
          <div
            className="col-md-3"
          >
            <div
              className="new-customization-menu accordion"
            >
              <div
                className="card"
              >
                <div
                  className="card-header"
                >
                  <div
                    className="card-header"
                    onClick={[Function]}
                    variant="link"
                  >
                    Theme
                  </div>
                </div>
                <div
                  aria-expanded={null}
                  className="collapse show"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="flex-column nav card-header-undefined"
                      onKeyDown={[Function]}
                    >
                      <a
                        aria-current={null}
                        href="/console/theming/preview-and-colors"
                        onClick={[Function]}
                      >
                        Preview & colors
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/logos"
                        onClick={[Function]}
                      >
                        Logos
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/buttons"
                        onClick={[Function]}
                      >
                        Buttons
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/navigation"
                        onClick={[Function]}
                      >
                        Navigation
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/footer"
                        onClick={[Function]}
                      >
                        Footer
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/hero"
                        onClick={[Function]}
                      >
                        Hero
                      </a>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="card"
              >
                <div
                  className="card-header"
                >
                  <div
                    className="card-header"
                    onClick={[Function]}
                    variant="link"
                  >
                    Custom Pages
                  </div>
                </div>
                <div
                  aria-expanded={null}
                  className="collapse"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="flex-column nav card-header-undefined"
                      onKeyDown={[Function]}
                    >
                      <a
                        aria-current={null}
                        href="/console/custom-pages/about"
                        onClick={[Function]}
                      >
                        About
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/contact"
                        onClick={[Function]}
                      >
                        Contact
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/donate"
                        onClick={[Function]}
                      >
                        Donate
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/tos"
                        onClick={[Function]}
                      >
                        Terms of Service
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/honor"
                        onClick={[Function]}
                      >
                        Honor Code
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/privacy"
                        onClick={[Function]}
                      >
                        Privacy Policy
                      </a>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="card"
              >
                <div
                  className="card-header"
                >
                  <div
                    className="card-header"
                    onClick={[Function]}
                    variant="link"
                  >
                    Instance Settings
                  </div>
                </div>
                <div
                  aria-expanded={null}
                  className="collapse"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="flex-column nav card-header-undefined"
                      onKeyDown={[Function]}
                    >
                      <a
                        aria-current={null}
                        href="/console/settings/general"
                        onClick={[Function]}
                      >
                        General
                      </a>
                      <a
                        aria-current="page"
                        className="disabled active"
                        href="/"
                        onClick={[Function]}
                        style={Object {}}
                      >
                        Domain
                      </a>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="card"
              >
                <div
                  className="card-header"
                >
                  <div
                    className="card-header"
                    onClick={[Function]}
                    variant="link"
                  >
                    Courses (Studio)
                  </div>
                </div>
                <div
                  aria-expanded={null}
                  className="collapse"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="flex-column nav card-header-undefined"
                      onKeyDown={[Function]}
                    >
                      <a
                        aria-current={null}
                        href="/console/courses/manage"
                        onClick={[Function]}
                      >
                        Manage
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col-md-7"
          >
            <div
              className="preview-header d-flex align-items-center"
            >
              <div
                className="dropdown-container d-flex flex-column align-items-center"
              >
                <a
                  className="nav-link"
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="button"
                >
                  Dashboard
                  <i
                    className="fas xs fa-chevron-down"
                  />
                </a>
                <span
                  className="fill-line"
                />
                <div
                  className="dropdown-items align-items-center"
                >
                  <div
                    className="content d-flex flex-column"
                  >
                    <div
                      aria-expanded={null}
                      className="collapse"
                    >
                      <button
                        className="d-inline-block"
                        onClick={[Function]}
                        type="button"
                      >
                        <p>
                          Dashboard
                        </p>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="notice-container d-flex flex-row-reverse"
              >
                <p>
                  This is only a preview. Publish changes to update the live site
                </p>
              </div>
            </div>
            <div
              className="preview-box card"
            >
              <div
                className="customization-form"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Theme preview page renders without crashing 1`] = `
<div
  className="console-page"
>
  <div
    className="d-flex justify-content-center align-middle redeployment-toolbar"
  >
    <div
      className="redeployment-nav"
    >
      <span
        className="status-pill badge badge-pill badge-primary"
        style={
          Object {
            "backgroundColor": "grey",
          }
        }
      >
        <div
          className="text"
        >
          Loading
        </div>
        <i
          className="fas fa-question-circle fa-lg"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        />
      </span>
      <button
        className="float-right btn-disabled btn btn-primary btn-lg"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        <div
          className="button-text-layer"
        >
          Publish changes
        </div>
      </button>
    </div>
    <div
      className="toast-container"
    />
  </div>
  <div
    className="new-console-page-container"
  >
    <div
      className="new-console-page-content row"
    >
      <div
        className="container-fluid"
      >
        <div
          className="justify-content-center row"
        >
          <div
            className="col-md-3"
          >
            <div
              className="new-customization-menu accordion"
            >
              <div
                className="card"
              >
                <div
                  className="card-header"
                >
                  <div
                    className="card-header"
                    onClick={[Function]}
                    variant="link"
                  >
                    Theme
                  </div>
                </div>
                <div
                  aria-expanded={null}
                  className="collapse show"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="flex-column nav card-header-undefined"
                      onKeyDown={[Function]}
                    >
                      <a
                        aria-current={null}
                        href="/console/theming/preview-and-colors"
                        onClick={[Function]}
                      >
                        Preview & colors
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/logos"
                        onClick={[Function]}
                      >
                        Logos
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/buttons"
                        onClick={[Function]}
                      >
                        Buttons
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/navigation"
                        onClick={[Function]}
                      >
                        Navigation
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/footer"
                        onClick={[Function]}
                      >
                        Footer
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/hero"
                        onClick={[Function]}
                      >
                        Hero
                      </a>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="card"
              >
                <div
                  className="card-header"
                >
                  <div
                    className="card-header"
                    onClick={[Function]}
                    variant="link"
                  >
                    Custom Pages
                  </div>
                </div>
                <div
                  aria-expanded={null}
                  className="collapse"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="flex-column nav card-header-undefined"
                      onKeyDown={[Function]}
                    >
                      <a
                        aria-current={null}
                        href="/console/custom-pages/about"
                        onClick={[Function]}
                      >
                        About
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/contact"
                        onClick={[Function]}
                      >
                        Contact
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/donate"
                        onClick={[Function]}
                      >
                        Donate
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/tos"
                        onClick={[Function]}
                      >
                        Terms of Service
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/honor"
                        onClick={[Function]}
                      >
                        Honor Code
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/privacy"
                        onClick={[Function]}
                      >
                        Privacy Policy
                      </a>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="card"
              >
                <div
                  className="card-header"
                >
                  <div
                    className="card-header"
                    onClick={[Function]}
                    variant="link"
                  >
                    Instance Settings
                  </div>
                </div>
                <div
                  aria-expanded={null}
                  className="collapse"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="flex-column nav card-header-undefined"
                      onKeyDown={[Function]}
                    >
                      <a
                        aria-current={null}
                        href="/console/settings/general"
                        onClick={[Function]}
                      >
                        General
                      </a>
                      <a
                        aria-current="page"
                        className="disabled active"
                        href="/"
                        onClick={[Function]}
                        style={Object {}}
                      >
                        Domain
                      </a>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="card"
              >
                <div
                  className="card-header"
                >
                  <div
                    className="card-header"
                    onClick={[Function]}
                    variant="link"
                  >
                    Courses (Studio)
                  </div>
                </div>
                <div
                  aria-expanded={null}
                  className="collapse"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="flex-column nav card-header-undefined"
                      onKeyDown={[Function]}
                    >
                      <a
                        aria-current={null}
                        href="/console/courses/manage"
                        onClick={[Function]}
                      >
                        Manage
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col-md-7"
          >
            <div
              className="customization-form"
            >
              <div
                className="loading"
              >
                <i
                  className="fas fa-2x fa-sync-alt fa-spin"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
